libminifi/test/sql-tests/mocks/MockConnectors.cpp

```diff
@@ -0,0 +1,386 @@
+  tables_.clear();
+  std::ifstream file(file_path_);
+  std::string line;
+  ParsePhase phase = ParsePhase::NEW_TABLE;
+  std::string table_name;
+  std::vector<std::string> column_names;
+  std::vector<DataType> column_types;
+  while (std::getline(file, line)) {
+    switch (phase) {
+      case ParsePhase::NEW_TABLE: {
+        table_name = minifi::utils::StringUtils::trim(line);
+        phase = ParsePhase::COLUMN_NAMES;
+        break;
+      }
+      case ParsePhase::COLUMN_NAMES: {
+        column_names = minifi::utils::StringUtils::splitAndTrim(line, "|");
+        phase = ParsePhase::COLUMN_TYPES;
+        break;
+      }
+      case ParsePhase::COLUMN_TYPES: {
+        column_types.clear();
+        auto type_strs = minifi::utils::StringUtils::splitAndTrim(line, "|");
+        for (const auto& type : type_strs) {
+          column_types.push_back(stringToDataType(type));
+        }
+        tables_.emplace(table_name, MockRowset(column_names, column_types));
+        phase = ParsePhase::ROW_VALUES;
+        break;
+      }
+      case ParsePhase::ROW_VALUES: {
+        line = minifi::utils::StringUtils::trim(line);
+        if (line.empty()) {
+          phase = ParsePhase::NEW_TABLE;
+          break;
+        }
+        auto cells = minifi::utils::StringUtils::splitAndTrim(line, "|");
+        tables_.at(table_name).addRow(cells);
+        break;
+      }
+    }
+  }
+}
+
+void MockDB::storeDb() {
+  std::ofstream file(file_path_);
+  for (const auto& table : tables_) {
+    file << table.first << std::endl;
+    auto& rowset = table.second;
+    for (const auto& col_name : rowset.getColumnNames()) {
+      file << col_name << "|";
+    }
+    file << std::endl;
+    for (const auto& col_type : rowset.getColumnTypes()) {
+      file << dataTypeToString(col_type) << "|";
+    }
+    file << std::endl;
+    for (const auto& row : rowset.getRows()) {
+      for (const auto& cell : row.getValues()) {
+        file << cell << "|";
+      }
+      file << std::endl;
+    }
+    file << std::endl;
+  }
+}
+
+DataType MockDB::stringToDataType(const std::string& type_str) {
+  if (type_str == "integer") return DataType::INTEGER;
+  if (type_str == "text") return DataType::STRING;
+  if (type_str == "real") return DataType::DOUBLE;
+  throw std::runtime_error("Unimplemented data type");
+}
+
+std::string MockDB::dataTypeToString(DataType data_type) {
+  switch (data_type) {
+    case DataType::INTEGER: return "integer";
+    case DataType::STRING: return "text";
+    case DataType::DOUBLE: return "real";
+    default: throw std::runtime_error("Unimplemented data type");
+  }
+}
+
+std::unique_ptr<Rowset> MockStatement::execute(const std::vector<std::string>& args) {
+  MockDB db(file_path_);
+  return db.execute(query_, args);
+}
+
+MockODBCConnection::MockODBCConnection(std::string connectionString)
+    : connection_string_(std::move(connectionString)) {
+  std::smatch match;
+  std::regex expr("Database=(.*)(;|$)");
```

Comment:
not clear whether we need it or not, but there is a `Regex` class in `RegexUtils.h`, aiming to address GCC<4.9 compilers, should we use that here as well?

Reply:
My thinking was that as we are already on the way of dropping support for GCC 4.8 and we already have a ticket [MINIFICPP-1508](https://issues.apache.org/jira/browse/MINIFICPP-1508) for removing the `Regex` utility class it wouldn't be ideal to add more functionality to it.